src/AspNetCore.Identity.MongoDB/MongoIdentityUser.cs

```diff
@@ -33,7 +33,7 @@ public MongoIdentityUser(string userName)
                 throw new ArgumentNullException(nameof(userName));
             }
 
-            Id = GenerateId(userName);
+            // Id = GenerateId(userName); // Id will generate by mongodb driver
```

Comment:
OTH, we need to give it an within the ctor: `Id = ObjectId.GenerateNewId().ToString();` I will do that, too.